src/SubscriberBase.h

```diff
@@ -83,6 +83,15 @@ class SubscriberBaseT : public Subscriber<T>,
       bool startExecutor = true)
       : ExecutorBase(executor, startExecutor), cancelled_(false) {}
 
+  virtual ~SubscriberBaseT() {
+    if (!cancelled_ && originalSubscription_) {
```

Comment:
Not sure if I understand the question well. RS hold shared_pr to the Subscriber and makes sure it will call on{Error,Complete}. The implementation of that method will schedule on{Error,Complete}Impl to be executed on Executor and will capture thisPtr, a hard shared_ptr reference, in the lambda parameter.
Am I answering your question?